### PR TITLE
Drop full aws sdk

### PR DESCRIPTION
**Pull request by asolenzal**
issue: [https://github.com/articulate/identity/issues/947](https://github.com/articulate/identity/issues/947)

## Review steps
Described in this PR [https://github.com/articulate/heyokra/pull/393](https://github.com/articulate/heyokra/pull/393)

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+  if (queueUrl === undefined) {
+    throw Error(`SQS: Failed to create queue: ${queueName}`)
+  }
+
+  const queueArn = await getQueueArn(queueUrl)
+  if (queueArn === undefined) {
+    throw Error(`SQS: Failed to get arn for: ${queueUrl}`)
+  }
+
+  if (topicOptions.arn !== undefined && topicOptions.arn !== null && topicOptions.arn !== '') {
```
Comment (lucasadrianof) [inline src/index.ts:148]:
```suggestion
  if (topicOptions?.arn) {
```

File: test/_setup.ts
Comment (lucasadrianof) [inline test/_setup.ts:1]: Do we still need `nock` here? I think we could remove this file and that dependency, the `aws-sdk-client-mock` we're using already prevents the SDK from hitting the real AWS URL.

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    "sourceMap": true,
+    "target": "ES2018",
+    "moduleResolution": "nodenext",
+    "module": "CommonJS",
+    "strictNullChecks": true,
+    "esModuleInterop": true,
+  },
+  "include": ["./src/**/*.ts"],
+  "exclude": ["./test/**/*.ts"]
+}
```
Comment (lucasadrianof) [inline tsconfig.json:14]:
```suggestion
}

```

File: src/index.ts [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    const subscriptionArn = await createTopicSubscription(queueArn, topicOptions.arn)
+    await setSqsQueueAttributes(queueUrl, queueArn, topicOptions.arn)
+
+    if (topicOptions.filterPolicy !== null) {
```
Comment (lucasadrianof) [inline src/index.ts:152]:
```suggestion
    if (topicOptions.filterPolicy) {
```

File: src/index.ts [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    await createQueue({ QueueName: `${queueName}-DLQ` })
+      .then(queue => queue.QueueUrl)
+      .catch(() => undefined)
+
+  const getQueueArn = async (QueueUrl: string): Promise<string | undefined> =>
+    await getQueueAttributes({
+      QueueUrl,
+      AttributeNames: ['QueueArn']
+    })
+      .then((attributes: GetQueueAttributesCommandOutput) => attributes.Attributes?.QueueArn)
```
Comment (lucasadrianof) [inline src/index.ts:80]: Could we add one more test case covering the scenario when this attribute is not returned? I ran a quick coverage on this branch with your changes and this is the only bit not covered. If we add a test validating this we'll get 100% coverage 🥳 .

image


```suggestion
      .then((attributes) => attributes.Attributes?.QueueArn)
```

File: src/types.ts
Comment (lucasadrianof) [inline src/types.ts:1]: Could we alpha-sort the interfaces on this file? 🙇🏽‍♂️

File: package.json [hunk 5]
```diff
@@ -22,16 +28,31 @@
...
+    "@typescript-eslint/eslint-plugin": "^5.43.0",
+    "@typescript-eslint/parser": "^5.56.0",
     "aws-sdk-mock": "^4.5.0",
     "chai": "^4.2.0",
     "chai-as-promised": "^7.1.1",
+    "eslint": "^8.0.1",
+    "eslint-config-standard-with-typescript": "^34.0.1",
+    "eslint-plugin-import": "^2.25.2",
+    "eslint-plugin-n": "^15.0.0",
+    "eslint-plugin-promise": "^6.0.0",
```
Comment (lucasadrianof) [inline package.json:50]: Are we using these plugins? I don't see any references to them in the eslint config file.

**Review by lucasadrianof — CHANGES_REQUESTED** [review]
Dang, this is a really nice work, I love it 🥳 .

I really liked how we were able to move away from `ramda` and use the new AWS SDK with TS with no other extra dependencies. Nice addition with the eslint plugin too 👏🏽 .

I have a few suggestions, but overall this looks great!

**Comment by asolenzal** [comment]
@lucasadrianof All comments addressed, feel free to check again.

**Pull request merged**